src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -455,11 +455,14 @@ private static Connection getClosedConnection()
             public Object invoke(Object proxy, Method method, Object[] args) throws Throwable
             {
                final String methodName = method.getName();
-               if ("abort".equals(methodName)) { 
+               if ("isValid".equals(methodName)) {
+                  return Boolean.FALSE;
+               }
+               else if ("close".equals(methodName)) { 
```

Comment:
Why on earth would someone unwrap and call close?  BTW, they *can* call ``proxycon.unwrap().close()``.  But they cannot call:
```java
proxycon.close()
proxycon.unwrap().close()
```
This is a "use after return" to pool.  In almost every other pool the ``proxycon.unwrap().close()`` call would close the Connection *after a new thread has acquired it*!

I would actually prefer for HikariCP to fail in spectacular fashion with lots of fireworks in this case.  If you want to implement this method as:
```java
if ("close".equals(methodName)) { 
   throw new IllegalStateException("WTF are you doing?!");
}
```
then I have no objection.